src/astarte/handler.rs

```diff
@@ -321,6 +320,44 @@ impl DevicePublisher {
 
         Ok(())
     }
+
+    // check if the interface is present in the node introspection
+    async fn find_in_introspection(
+        &self,
+        node_id: NodeId,
+        interface_name: &str,
+    ) -> Result<(), AstarteMessageHubError> {
+        let node = self.subscribers.read().await;
```

Comment:
nit: since it should be a list of nodes.
```suggestion
        let nodes = self.subscribers.read().await;
```